### PR TITLE
feat: fixed minor bugs for parent issue fix

### LINKED ISSUE #268: Disable Bounties for "Parent Issues"
### Context

It's fast and easy to write todo lists in the original comment of a GitHub issue, and then immediately convert them to "child issues" directly from the GitHub user interface.

The problem I am seeing is that the bounty system breaks down where contributors can "double dip" on rewards easily by claiming bounties for the actual bounties (child issues) and then the "meta" or "parent" issue.

The argument possibly could be made for project management. However in the future all of the **issue creators**, and **commenters** (both are how the project manager would "manage") are to receive compensation for their contributions upon the completion of the bounty.

### Proposal

- When changing a label, the bot should check if there are linked issues in the original comment and disable setting a price (and most importantly, as a consequence, payment permit generation.)

- The bot should comment explaining why it refuses to set a price.

---

              Parent issues should not have bounties. Instead, the child issues should have bounties.

_Originally posted by @pavlovcik in https://github.com/ubiquity/grants/issues/5#issuecomment-1519067539_

### PR DESCRIPTION
**Pull request by wannacfuture**
closes #268 
fixed some concerns
https://github.com/wannacfuture/Battleship/issues/77

### CODE REVIEW

**Comment by wannacfuture** [comment]
hey, @web4er 
Just need QA be done for this PR.

File: src/configs/strings.ts [hunk 1]
```diff
@@ -2,4 +2,6 @@ export const GLOBAL_STRINGS = {
   unassignComment: "Releasing the bounty back to dev pool because the allocated duration already ended!",
   askUpdate: "Do you have any updates",
   assignCommandDisabledComment: "The `/assign` command is disabled for this repository.",
+  skipPriceLabelGenerationComment: "Price label generation skipped since it is identified as parent issue.",
```
Comment (pavlovcik) [inline src/configs/strings.ts:5]:
```suggestion
  skipPriceLabelGenerationComment: "Pricing is disabled on parent issues.",
```

File: src/configs/strings.ts [hunk 2]
```diff
@@ -2,4 +2,6 @@ export const GLOBAL_STRINGS = {
...
+  ignorStartCommandForParentIssuesComment: "The `/start` command is disabled for parent issues.",
```
Comment (pavlovcik) [inline src/configs/strings.ts:6]:
```suggestion
  ignoreStartCommandForParentIssueComment: "Please select a child issue from the specification checklist to work on. The `/start` command is disabled on parent issues.",
```

**Review by pavlovcik — CHANGES_REQUESTED** [review]
Minor adjustments needed.

**Comment by wannacfuture** [comment]
REady for QA

**Review by pavlovcik — APPROVED** [review]
Code looks good.

**Comment by pavlovcik** [comment]
> Ready for QA

I accidentally merged it but lets just handle QA retroactively then (you can do a hot fix if QA finds an issue)

**Comment by web4er** [comment]
> hey, @web4er Just need QA be done for this PR.

QA: https://github.com/web4erOrg/ubiquibot/issues/20

Failures:
- if user opens a PR, and PR is merged, permnit is being generated. Permit should not be generated. https://github.com/web4erOrg/ubiquibot/issues/20#issuecomment-1672879797
- bot is adding `pricing is disabled...` message multiple times at start. Can be seen both in https://github.com/web4erOrg/ubiquibot/issues/20 and also was reproduced in https://github.com/web4erOrg/ubiquibot/issues/24
- some price labels are not being removed by the bot https://github.com/web4erOrg/ubiquibot/issues/20#issuecomment-1672897384

**Comment by wannacfuture** [comment]
I fixed the other things. But multiple message issue:

You added both Time label and priority label at once so the pricingLabelLogic is called twice.
that's the reason why the warning message shows twice.

Rfc, @pavlovcik , @web4er

**Comment by wannacfuture** [comment]
> I thought we programmed the comment function to edit the last reply to prevent double posts anyways?

Not sure about this. Could you provide the link or something?

**Comment by pavlovcik** [comment]
- https://github.com/ubiquity/ubiquibot/issues/440

**Comment by web4er** [comment]
> > I thought we programmed the comment function to edit the last reply to prevent double posts anyways?
> 
> Not sure about this. Could you provide the link or something?

https://github.com/ubiquity/ubiquibot/blob/2abac46f6ecba351677f528f2fa3b29e200029b3/src/helpers/issue.ts#L149
There is `upsertComment` function instead of `addComment` function. 

But it has a condition
https://github.com/ubiquity/ubiquibot/blob/2abac46f6ecba351677f528f2fa3b29e200029b3/src/helpers/issue.ts#L150

**Comment by pavlovcik** [comment]
> https://github.com/ubiquity/ubiquibot/blob/2abac46f6ecba351677f528f2fa3b29e200029b3/src/helpers/issue.ts#L150


Okay so if I am understanding this correctly, this means that if the human edits their comment, that the bot will edit its comment. 

How about a hot fix so that the bot will never double post? Does that make sense? It will band-aid a lot of the small edge cases like when it aggressively follows up a ton of times, or what we were discussing earlier in this conversation.

**Comment by web4er** [comment]
> Okay so if I am understanding this correctly, this means that if the human edits their comment, that the bot will edit its comment.

correct

> How about a hot fix so that the bot will never double post? Does that make sense? It checks the last comment in the conversation, and if it is the last comment, it will edit it instead of double posting.

For this PR, In a new hotfix https://github.com/ubiquity/ubiquibot/pull/598 the initial double comment about pricing has been changed into no comment at all. And I have already approved it. I think in most cases user already knows that pricing can't be set in parent issue, and it is like noise even as a single comment. In case user doesn't know, the `Pricing is disabled on parent issues.` will show up if user tries to set price label. 

> It will band-aid a lot of the small edge cases like when it aggressively follows up a ton of times, or what we were discussing earlier in this conversation.

I think those follow-ups have different causes and the root cause should be fixed. Editing the bot comment will be like sweeping them under the rug.

**Pull request merged**